### ISSUE TITLE
[Tune/CI] SigOpt Tests are skipped in CI

### ISSUE THREAD

**Issue by Yard1** [ray-project/ray#26567]
### What happened + What you expected to happen

Tune unit tests concerning SigOpt searcher are currently skipped in CI (in code) and the docs notebook is excluded from CI explicitly. This is due to the missing SigOpt API key in the CI environments. Anyscale has its own API key, and we would like to use it without leaking it. One way to do it would be to run the tests & notebook only for branch builds, like wandb and comet tests.

### Versions / Dependencies

master

### Reproduction script

-

### Issue Severity

Low: It annoys or frustrates me.

**Comment by richardliaw** [comment]
Can you pass it in via the buildkite UI? cc @simon-mo

**Comment by simon-mo** [comment]
No. The secrets must be injected through our CI infra manually. Release tests is the easiest way to put in such secret; CI build it a bit tough
